Tentacle/RepositoryInfo.swift

```diff
@@ -0,0 +1,99 @@
+//
+//  RepositoryInfo.swift
+//  Tentacle
+//
+//  Created by Romain Pouclet on 2016-08-02.
+//  Copyright © 2016 Matt Diephouse. All rights reserved.
+//
+
+import Foundation
+import Argo
+import Curry
+
+public struct RepositoryInfo: Hashable, CustomStringConvertible {
+    /// The id of the repository
+    public let id: String
+    
+    /// The basic informations about the owner of the repository, either an User or an Organization
+    public let owner: Owner
```

Comment:
But it can also be an Organization, that's why the Owner has a type property. I think it also has different properties than a regular User object. 
I can't really check right now since I'm on my phone but do you think there is a way to get rid of the Owner Struct?